File: app-laravel/src/Order/Application/Queries/ListOrders/ListOrdersHandler.php
<?php

declare(strict_types=1);

namespace Src\Order\Application\Queries\ListOrders;

use Src\Order\Application\DTOs\OrderDTO;
use Src\Order\Infrastructure\Models\OrderModel;
use Src\Shared\Application\Bus\Query;
use Src\Shared\Application\Bus\QueryHandler;

/**
 * LIST ORDERS HANDLER (CQRS Pattern)
 * ====================================
 * ListOrdersQuery-ni emal edib səhifələnmiş OrderDTO siyahısı qaytarır.
 *
 * Bu handler istifadəçinin sifarişlərini səhifələyərək tapır və DTO-ya çevirir.
 * Heç sifariş yoxdursa, boş səhifələnmiş nəticə qaytarır (xəta vermir).
 *
 * Səhifələmə (pagination):
 * - perPage: Hər səhifədəki element sayı (standart: 10)
 * - page: Cari səhifə nömrəsi
 */
class ListOrdersHandler implements QueryHandler
{
    public function __construct() {}

    /**
     * @param Query $query ListOrdersQuery olmalıdır
     * @return \Illuminate\Contracts\Pagination\LengthAwarePaginator Səhifələnmiş nəticə
     */
    public function handle(Query $query): mixed
    {
        /** @var ListOrdersQuery $query */

        // İstifadəçinin sifarişlərini səhifələnmiş şəkildə alırıq
        $paginated = OrderModel::forUser($query->userId())->paginate(
            perPage: $query->perPage(),
            page: $query->page(),
        );

        // Hər Order modelini DTO-ya çeviririk, səhifələmə məlumatını saxlayırıq
        $paginated->getCollection()->transform(
            fn ($order) => OrderDTO::fromModel($order),
        );

        return $paginated;
    }
}

File: app-spring/src/main/java/az/ecommerce/order/application/query/listorders/ListOrdersHandler.java
package az.ecommerce.order.application.query.listorders;

import az.ecommerce.order.application.dto.OrderDto;
import az.ecommerce.order.domain.repository.OrderRepository;
import az.ecommerce.shared.application.bus.QueryHandler;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ListOrdersHandler implements QueryHandler<ListOrdersQuery, List<OrderDto>> {
    private final OrderRepository repository;
    public ListOrdersHandler(OrderRepository r) { this.repository = r; }
    @Override public List<OrderDto> handle(ListOrdersQuery q) {
        return repository.findByUserId(q.userId()).stream().map(OrderDto::fromDomain).toList();
    }
}
